src/nvim/tempfile.c

```diff
@@ -55,28 +55,48 @@ static void vim_maketempdir(void)
   }
 }
 
-/// Delete the temp directory and all files it contains.
-void vim_deltempdir(void)
+/// Delete "name" and everything in it, recursively.
+/// @param name The path which should be deleted.
+/// @return 0 for success, -1 if some file was not deleted.
+int delete_recursive(char_u *name)
```

Comment:
This should probably go in fileio.c, but unfortunately fileio.c includes `tempfile.h`. 

I'm not sure that breaking out tempfile-related functions into tempfile.h really buys us much: 

- it's 138 lines
- it is application-specific logic (otherwise it would live in `os/*`)
- it's pretty relevant to `fileio.c`

So it should probably be moved back into `fileio.c`.